### PR TITLE
add uperf job to the pipeline

### PR DESCRIPTION
**Pull request by mohit-sheth**
@chaitanyaenr @jtaleric 
Adds uperf test to the jenkins pipeline

### CODE REVIEW

File: jjb/dynamic/uperf.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    builders:
+    - shell: |+
+        set -o pipefail
+        set -eux
+        # Disable logging
+        set +x
+        oc login -u ${CLUSTER_USER} -p ${CLUSTER_PASSWORD} ${CLUSTER_API_URL}
+        # Run tests
+        set -x
+        git clone https://github.com/cloud-bulldozer/plow.git
```
Comment (chaitanyaenr) [inline jjb/dynamic/uperf.yml]: We can skip this as we are cloning the plow repo using the github jenkins plugin ( recommended way ) in line 96.

File: pipeline-scripts/uperf.groovy [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+						[$class: 'StringParameterValue', name: 'CLUSTER_PASSWORD', value: cluster_password ],
+						[$class: 'StringParameterValue', name: 'CLUSTER_API_URL', value: cluster_api_url ],
+						[$class: 'StringParameterValue', name: 'ES_SERVER', value: es_server ],
+						[$class: 'StringParameterValue', name: 'ES_PORT', value: es_port ],
+						[$class: 'BooleanParameterValue', name: 'METADATA_COLLECTION', value: Boolean.valueOf(metadata_collection) ],
+			} catch ( Exception e) {
+				echo "UPERF Job failed with the following error: "
+				echo "${e.getMessage()}"
+				mail(
+					to: 'nelluri@redhat.com',
```
Comment (chaitanyaenr) [inline pipeline-scripts/uperf.groovy]: Add yourself and Joe's email id. @jtaleric should we create a mailing list for receiving the email like you suggested in the trigger PR?

File: pipeline-scripts/uperf.groovy [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			def cluster_password = uperf_properties['CLUSTER_PASSWORD']
+			def cluster_api_url = uperf_properties['CLUSTER_API_URL']
+			def es_server = uperf_properties['ES_SERVER']
+			def es_port = uperf_properties['ES_PORT']
+			def metadata_collection = uperf_properties['METADATA_COLLECTION']
+			
+			try {
+				uperf_build = build job: 'RIPSAW-UPERF',
+				parameters: [   [$class: 'LabelParameterValue', name: 'node', label: node_label ],
+                                                [$class: 'BooleanParameterValue', name: 'SKIP_TLS_VERIFICATION', value: Boolean.valueOf(skip_tls) ],
```
Comment (chaitanyaenr) [inline pipeline-scripts/uperf.groovy]: tabs vs spaces issue

File: pipeline-scripts/uperf.groovy [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		node(node_label) {
+			// get properties file
+			if (fileExists(property_file_name)) {
+				println "Looks like the property file already exists, erasing it"
+				sh "rm ${property_file_name}"
+			}
+			// get properties file
+			sh "wget ${UPERF_PROPERTY_FILE} -O ${property_file_name}"
+			sh "cat ${property_file_name}"
+                        def uperf_properties = readProperties file: property_file_name
```
Comment (chaitanyaenr) [inline pipeline-scripts/uperf.groovy]: tabs vs spaces issue

File: jjb/dynamic/uperf.yml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    - string:
+        default: ""
+        description: Password for CLUSTER_USER to access cluster.
+        name: CLUSTER_PASSWORD
+    - string:
+        default: ""
+        description: The URL address to the openshift cluster to login to.
+        name: CLUSTER_API_URL
+    - string:
+        default: "search-cloud-perf-lqrf3jjtaqo7727m7ynd2xyt4y.us-west-2.es.amazonaws.com"
```
Comment (chaitanyaenr) [inline jjb/dynamic/uperf.yml]: Let's set the default to null and avoid exposing the var.

File: pipeline-scripts/uperf.groovy [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+						[$class: 'StringParameterValue', name: 'CLUSTER_PASSWORD', value: cluster_password ],
+						[$class: 'StringParameterValue', name: 'CLUSTER_API_URL', value: cluster_api_url ],
+						[$class: 'StringParameterValue', name: 'ES_SERVER', value: es_server ],
+						[$class: 'StringParameterValue', name: 'ES_PORT', value: es_port ],
+						[$class: 'BooleanParameterValue', name: 'METADATA_COLLECTION', value: Boolean.valueOf(metadata_collection) ]]
+			} catch ( Exception e) {
+				echo "UPERF Job failed with the following error: "
+				echo "${e.getMessage()}"
+				mail(
+					to: 'jtaleric@redhat.com', 'msheth@redhat.com'
```
Comment (chaitanyaenr) [inline pipeline-scripts/uperf.groovy]: missing a comma after the to address

File: jjb/dynamic/uperf.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
+- job:
+    block-downstream: false
+    block-upstream: false
+    builders:
+    - shell: |+
+        set -o pipefail
+        set -eux
+        # Disable logging
+        set +x
+        oc login -u ${CLUSTER_USER} -p ${CLUSTER_PASSWORD} ${CLUSTER_API_URL}
```
Comment (chaitanyaenr) [inline jjb/dynamic/uperf.yml]: This needs to support skipping the tls verification similar to https://github.com/openshift-scale/scale-ci-pipeline/blob/master/jjb/dynamic/scale-ci_nodevertical.yml#L11.

File: jjb/dynamic/uperf.yml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        # Run tests
+        set -x
+        pushd plow/workloads/network-perf/
```
Comment (chaitanyaenr) [inline jjb/dynamic/uperf.yml]: This github jenkins plugin clones the content in the workspace directory without the repo name, so this needs to "pushd workloads/network-perf"

File: jjb/dynamic/uperf.yml [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+          oc login -u ${CLUSTER_USER} -p ${CLUSTER_PASSWORD} ${CLUSTER_API_URL} --insecure-skip-tls-verify
+        else
+          oc login -u ${CLUSTER_USER} -p ${CLUSTER_PASSWORD} ${CLUSTER_API_URL}
+        fi
+        # Run tests
+        set -x
+        pushd workloads/network-perf
+        ./run_hostnetwork_network_test_fromgit.sh
+        ./run_pod_network_test_fromgit.sh
+        ./run_serviceip_network_test_fromgit.sh
```
Comment (chaitanyaenr) [inline jjb/dynamic/uperf.yml]: @mohit-sheth @jtaleric Should we call the network_compare script to determine pass/fail here or maybe we can add it to the hostnetwork/pod_network/serviceip network test scripts in plow?

Comment (jtaleric) [reply jjb/dynamic/uperf.yml]: IMHO leave the compare outside of this. 

https://github.com/cloud-bulldozer/plow/blob/master/workloads/network-perf/run_network_compare.sh

Should help with this.

Comment (jtaleric) [reply jjb/dynamic/uperf.yml]: We need to also pass a "baseline_uperf_uuid"

File: jjb/dynamic/uperf.yml [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        description: The elasticsearch server where the data will be indexed.
+        name: ES_SERVER
+    - string:
+        default: ""
+        description: The port for the elasticsearch server.
+        name: ES_PORT
+    - bool:
+        default: true
+        description: Enables/Disables metadata collection
+        name: METADATA_COLLECTION
```
Comment (jtaleric) [inline jjb/dynamic/uperf.yml]: We should include a `BASELINE_UPERF_UUID`, which we run our compare against

Comment (mohit-sheth) [reply jjb/dynamic/uperf.yml]: done

**Comment by aakarshg** [comment]
nice work @mohit-sheth

**Pull request merged**